wolfcrypt/src/port/Espressif/esp_crt_bundle/esp_crt_bundle.c

```diff
@@ -0,0 +1,1380 @@
+    int ret = 0;
+
+    wc_InitDecodedCert(&cert, der_cert, sz, NULL);
+
+    ret = wc_ParseCert(&cert, CERT_TYPE, NO_VERIFY, 0);
+
+    /* Check the special case of parse error with strict checking. */
+    if ((cert.serialSz == 1) && (cert.serial[0] == 0x0)) {
+        /* If we find a zero serial number, a parse error may still occur. */
+        if (ret == ASN_PARSE_E) {
+            /* Issuer amd subject will only be non-blank with relaxed check */
+            ESP_LOGW(TAG, "Encountered ASN Parse error with zero serial");
+            ESP_LOGCBI(TAG, "Issuer: %s", cert.issuer);
+            ESP_LOGCBI(TAG, "Subject: %s", cert.subject);
+#if defined(CONFIG_WOLFSSL_NO_ASN_STRICT) || \
+    defined(       WOLFSSL_NO_ASN_STRICT)
+            ESP_LOGW(TAG, "WOLFSSL_NO_ASN_STRICT enabled. Ignoring error.");
+
+            /* We'll force the return result to one for a "valid"
+             * parsing result, but not strict and found zero serial num. */
+            ret = 1;
+#else
+    #if defined(CONFIG_WOLFSSL_ASN_ALLOW_0_SERIAL) || \
+        defined(       WOLFSSL_ASN_ALLOW_0_SERIAL)
+            /* Issuer amd subject will only be non-blank with relaxed check */
+            ESP_LOGCBW(TAG, "WOLFSSL_ASN_ALLOW_0_SERIAL enabled. "
+                            "Ignoring error.");
+
+            /* We'll force the return result for a "valid" parsing result,
+             * not strict and found zero serial num. */
+            ret = 1;
+    #else
+            ESP_LOGE(TAG, "ERROR: Certificate must have a Serial Number.");
+            ESP_LOGE(TAG, "Define WOLFSSL_NO_ASN_STRICT or "
+                          "WOLFSSL_ASN_ALLOW_0_SERIALto relax checks.");
+            /* ret (Keep ASN_PARSE_E)  */
+    #endif
+#endif
+        } /* special ASN_PARSE_E handling */
+        else {
+            /* Not an ASN Parse Error; the zero configured to be allowed. */
+            ESP_LOGV(TAG, "WARNING: Certificate has no Serial Number.");
+
+            /* If we found a zero, and the result of wc_ParseCert is zero,
+             * we'll return that zero as "cert has a zero serial number". */
+        }
+    }
+    else {
+        ESP_LOGV(TAG, "Not a special case zero serial number.");
+    }
+
+    if (ret > -1) {
+        ESP_LOGV(TAG, "Issuer: %s", cert.issuer);
+        ESP_LOGV(TAG, "Subject: %s", cert.subject);
+        ESP_LOGV(TAG, "Serial Number: %.*s", cert.serialSz, cert.serial);
+    }
+    else {
+        ESP_LOGCBV(TAG, "wolfssl_is_zero_serial_number exit  = %d", ret);
+    }
+
+    /* Clean up and exit */
+    wc_FreeDecodedCert(&cert);
+
+    return ret;
+}
+
+/* Public API wolfSSL_X509_get_cert_items() */
+int wolfSSL_X509_get_cert_items(char* CERT_TAG,
+                                WOLFSSL_X509* cert,
+                                WOLFSSL_X509_NAME** issuer,
+                                WOLFSSL_X509_NAME** subject)
+{
+    char stringVaue[X509_MAX_SUBJECT_LEN];
+#ifdef WOLFSSL_DEBUG_CERT_BUNDLE
+    char before_str[32];
```

Comment:
Maybe use `CTC_DATE_SIZE` instead of hard coded 32?

Reply:
Yes, good idea. added.